addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComInterfaceMessage.java

```diff
@@ -257,8 +301,27 @@ public void encodeMessage(byte[] data) throws RFXComException {
             enableARCPackets = (data[9] & 0x02) != 0;
             enableX10Packets = (data[9] & 0x01) != 0;
 
-            hardwareVersion1 = data[10];
-            hardwareVersion2 = data[11];
+            /**
+             * Different firmware versions have slightly different message formats.
+             * The firmware version numbering is unique to each hardware version
+             * but the location of the hardware version in the message is one of
+             * those things whose position varies. So we have to just look at the
+             * firmware version and pray. This condition below is taken from the
+             * openhab1-addons binding.
```

Comment:
Great that you ported this.

Please credit the author in both the commit message,using also-by and in the file using the @author tag.